client/src/main/java/io/pravega/client/connection/impl/RawClient.java

```diff
@@ -78,7 +79,11 @@ public void processingFailure(Exception error) {
         @Override
         public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenCheckFailed) {
             log.warn("Auth token check failed on segment {} with {}", segmentId, authTokenCheckFailed);
-            closeConnection(new AuthenticationException(authTokenCheckFailed.toString()));
+            if (authTokenCheckFailed.isTokenExpired()) {
+                closeConnection(new TokenExpiredException(authTokenCheckFailed.getServerStackTrace()));
```

Comment:
would this cause a change to the SegmentHelper logic at https://github.com/pravega/pravega/blob/4f7be177996bd5d3a05e2ab6f117d5a40b1e2207/controller/src/main/java/io/pravega/controller/server/SegmentHelper.java#L624-L641 ?

Reply:
Thanks. I've pushed the fix in the latest commit.